libipl/p10/common.C

```diff
@@ -237,6 +237,24 @@ bool ipl_check_functional_master(void)
 	return true;
 }
 
+struct pdbg_target * ipl_get_functional_master(void)
```

Comment:
you can modify bool ipl_is_master_proc(struct pdbg_target *proc) to support this ? by adding additional default parameter to check functional status inside the function.

File scope: Avoid usage of master name and replace with primary in all newly added code.

Reply:
here the requirement is to return early if not master proc is not present for which we need to loop through all the procs, as suggested in the code review comment earlier.

We are now adding a method that will return the primary processor, this is done to avoid 2 "for" loops.

will change method to primary processor

Reply:
Sorry , the function i mentioned earlier was related to isprimary  check , you requirement is get primary processor details,